### PR TITLE
Removed TypedAction and split RestCallback

### PR DESCRIPTION
**Pull request by meriouma**

### CODE REVIEW

File: gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/client/FailureCallback.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.gwtplatform.dispatch.rest.client;
```
Comment (olafleur) [inline gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/client/FailureCallback.java]: copyright notice

File: gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/client/ResponseCallback.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.gwtplatform.dispatch.rest.client;
```
Comment (olafleur) [inline gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/client/ResponseCallback.java]: idem

File: gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/client/SuccessCallback.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.gwtplatform.dispatch.rest.client;
```
Comment (olafleur) [inline gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/client/SuccessCallback.java]: idem

File: gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/client/interceptor/RestInterceptedAsyncCallback.java [hunk 4]
```diff
@@ -75,4 +81,32 @@ public void onSuccess(RestInterceptor result) {
...
+    }
+
+    protected void handleFailure(Throwable caught) {
+        dispatchRequest.cancel();
+
+        dispatchCall.onExecuteFailure(caught, null);
+    }
+
+    protected void delegateFailure(RestInterceptor interceptor) {
+        // TODO : Fix this
```
Comment (olafleur) [inline gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/client/interceptor/RestInterceptedAsyncCallback.java]: maybe write what is to be fixed?

**Comment by olafleur** [comment]
LGTM

File: gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/client/interceptor/RestInterceptedAsyncCallback.java (hunk 4 above)
Comment (meriouma) [inline gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/client/interceptor/RestInterceptedAsyncCallback.java]: Woops, this was more a reminder for me to fix before pushing :P

File: gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/client/ResponseCallback.java [hunk 5]
```diff
@@ -14,23 +14,11 @@
...
-        this.actionType = actionType;
-    }
+import com.google.gwt.http.client.Response;
 
-    @Override
-    public Class<A> getActionType() {
-        return actionType;
-    }
+@FunctionalInterface
+public interface ResponseCallback {
```
Comment (Chris-V) [inline gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/client/ResponseCallback.java]: AlwaysCallback?

File: gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/client/ResponseCallback.java [hunk 6]
```diff
@@ -14,23 +14,11 @@
...
+    void setResponse(Response response);
```
Comment (Chris-V) [inline gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/client/ResponseCallback.java]: onResponse(), execute(), always()?

File: gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/client/core/RestDispatchCall.java [hunk 7]
```diff
@@ -43,13 +42,19 @@
...
     private final CookieManager cookieManager;
     private final ResponseDeserializer responseDeserializer;
     private final RestInterceptorRegistry interceptorRegistry;
     private final RestDispatchHooks dispatchHooks;
+    private final A action;
+    private final RestCallback<R> callback;
+
+    private boolean intercepted;
 
     // TODO: Too many dependencies, split the logic
```
Comment (Chris-V) [inline gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/client/core/RestDispatchCall.java]: Gotta look at this soon I guess

**Comment by Chris-V** [comment]
LGTM

**Pull request merged**